Automattic/wp-calypso
# Update e2e tests to work with the new Pro plan

#### Changes proposed in this Pull Request

* This creates a new e2e test that targets the WordPress Pro plan.

#### Testing instructions

For testing this locally:
* [Review the documentation](https://wpcalypso.wordpress.com/devdocs/test/e2e/docs/tests_local.md) for testing locally to make sure you're set up correctly
* Add the feature flag `"plans/pro-plan": true` to your local config file.
* Create a new WP account and add the credentials to your local config file:
* * The new account should go under `testAccounts` and should be called `testUser`; this is referenced in the pro__purchase.ts file
* In the terminal, under `wp-calypso/tests/e2e` use this command: `yarn jest plans/pro__purchase.ts`

#### Still To Do
The current plan page and the new Pro plan have slightly different markup, so we can't just use the same selectors for the action button. I've been trying to resolve this by conditionally changing the selector depending on whether we should be showing the Pro plan or not. The best method might be to use the `isEligibleForProPlan()` conditional, which triggers other parts of the Pro plan code. This method currently returns false for `isE2ETest()` so we'll need to remove that check.

I'm 100% convinced there's a much better way to do this.
